include/stack_macros.h

```diff
@@ -66,81 +66,172 @@
-            ( pulStack[ 1 ] != ulCheckValue ) ||                                                 \
-            ( pulStack[ 2 ] != ulCheckValue ) ||                                                 \
-            ( pulStack[ 3 ] != ulCheckValue ) )                                                  \
-        {                                                                                        \
-            char * pcOverflowTaskName = pxCurrentTCB->pcTaskName;                                \
-            vApplicationStackOverflowHook( ( TaskHandle_t ) pxCurrentTCB, pcOverflowTaskName );  \
-        }                                                                                        \
-    } while( 0 )
+    #if( configNUMBER_OF_CORES == 1 )
+
+        #define taskCHECK_FOR_STACK_OVERFLOW()                                                       \
+        do                                                                                           \
+        {                                                                                            \
+            const uint32_t * const pulStack = ( uint32_t * ) pxCurrentTCB->pxStack;                  \
+            const uint32_t ulCheckValue = ( uint32_t ) 0xa5a5a5a5U;                                  \
+                                                                                                     \
+            if( ( pxCurrentTCB->pxTopOfStack <= pxCurrentTCB->pxStack + portSTACK_LIMIT_PADDING ) || \
+                ( pulStack[ 0 ] != ulCheckValue ) ||                                                 \
+                ( pulStack[ 1 ] != ulCheckValue ) ||                                                 \
+                ( pulStack[ 2 ] != ulCheckValue ) ||                                                 \
+                ( pulStack[ 3 ] != ulCheckValue ) )                                                  \
+            {                                                                                        \
+                char * pcOverflowTaskName = pxCurrentTCB->pcTaskName;                                \
+                vApplicationStackOverflowHook( ( TaskHandle_t ) pxCurrentTCB, pcOverflowTaskName );  \
+            }                                                                                        \
+        } while( 0 )
+
+    #else
+
+        #define taskCHECK_FOR_STACK_OVERFLOW( xCoreID )                                         \
+        do                                                                                      \
+        {                                                                                       \
+            const uint32_t * const pulStack = ( uint32_t * ) pxCurrentTCB->pxStack;             \
+            const uint32_t ulCheckValue = ( uint32_t ) 0xa5a5a5a5U;                             \
+            const TCB_t * const pxTCB = pxCurrentTCBs[ xCoreID ];                               \
```

Comment:
Should we set `pulStack` from `pxTCB` instead of `pxCurrentTCB`?

```suggestion
            const TCB_t * const pxTCB = pxCurrentTCBs[ xCoreID ];                               \
            const uint32_t * const pulStack = ( uint32_t * ) pxTCB->pxStack;                    \
            const uint32_t ulCheckValue = ( uint32_t ) 0xa5a5a5a5U;                             \
```

Reply:
Thank you for catching this @jefftenney !